source/array_list.c

```diff
@@ -184,9 +184,12 @@ static void aws_array_list_mem_swap(void *AWS_RESTRICT item1, void *AWS_RESTRICT
     }
 
     size_t remainder = item_size & (SLICE - 1); /* item_size % SLICE */
```

Comment:
Why not actually use `item_size % SLICE` here?  It's more understandable.  I know it's not part of the PR, just a comment.

Reply:
I believe it is because bitwise AND is faster than modulo, but I couldn't find a good reference.

Reply:
+1 on just using `%` instead of a tricky trick

this operation is just happening once, it's not in some tight loop where the optimization is going to make any difference